### PR TITLE
#42 add deck definition

### PR DESCRIPTION
**Pull request by Leo-gp**

### CODE REVIEW

File: Assets/Scripts/main/repository/Card Management/Deck Definition/ResourceDeckDefinitionRepository.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        protected abstract string ResourcePath { get; }
+        
+        public DeckDefinition Load()
+        {
+            return LoadFromResources();
+        }
+        
+        private DeckDefinition LoadFromResources()
+        {
+            LocalizationSettings.SelectedLocale = LocalizationSettings.ProjectLocale; // TODO: Remove
```
Comment (GingaaBread) [inline Assets/Scripts/main/repository/Card Management/Deck Definition/ResourceDeckDefinitionRepository.cs]: What is this? :)

File: Assets/Scripts/main/service/Card Management/CardPoolService.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public class CardPoolService : Service
         /// <summary>
         ///     The card pool entity containing discovered / available cards
         /// </summary>
-        private readonly CardPool _cardPool = new();
+        //private readonly CardPool _cardPool = new();
```
Comment (GingaaBread) [inline Assets/Scripts/main/service/Card Management/CardPoolService.cs:19]: Where is the new card pool?

**Review by GingaaBread — CHANGES_REQUESTED** [review]

File: Assets/Scripts/main/repository/Card Management/Deck Definition/ResourceDeckDefinitionRepository.cs (hunk 1 above)
Comment (Leo-gp) [reply Assets/Scripts/main/repository/Card Management/Deck Definition/ResourceDeckDefinitionRepository.cs]: This is from the first commit of this PR, it was just for testing purposes, it's not there anymore.

File: Assets/Scripts/main/service/Card Management/CardPoolService.cs (hunk 2 above)
Comment (Leo-gp) [reply Assets/Scripts/main/service/Card Management/CardPoolService.cs:19]: My bad, I forgot to uncomment stuff here, I'll keep it as it was (and `GameTests ` will pass again). Thanks for pointing out. I may do some rework in `CardPoolService` and `DeckService` for the next task, given the changes on this PR. With that I'll also remove the deck definition legacy classes (as I renamed them).

**Review by GingaaBread — APPROVED** [review]

**Pull request merged**